tests/handlers/tests.py

```diff
@@ -38,7 +38,8 @@ def test_bad_path_info(self):
         environ['PATH_INFO'] = b'\xed' if six.PY2 else '\xed'
         handler = WSGIHandler()
         response = handler(environ, lambda *a, **k: None)
-        self.assertEqual(response.status_code, 400)
+        # The path of the request will be encoded to '/%ED'.
```

Comment:
I confirmed that the bug fixed in 12265410 can be reintroduced because no tests are hitting the `except UnicodeDecodeError` block in the comment above. I think we need a new test or maybe that exception catching is obsolete?